src/main/java/io/confluent/connect/jdbc/source/TimestampIncrementingTableQuerier.java

```diff
@@ -229,6 +202,56 @@ public SourceRecord extractRecord() throws SQLException {
     return new SourceRecord(partition, offset.toMap(), topic, record.schema(), record);
   }
 
+  // Visible for testing
+  TimestampIncrementingOffset extractOffset(Schema schema, Struct record) {
+    final Timestamp extractedTimestamp;
+    if (timestampColumn != null) {
+      extractedTimestamp = (Timestamp) record.get(timestampColumn);
+      Timestamp timestampOffset = offset.getTimestampOffset();
+      assert timestampOffset != null && timestampOffset.compareTo(extractedTimestamp) <= 0;
+    } else {
+      extractedTimestamp = null;
+    }
+
+    final Long extractedId;
+    if (incrementingColumn != null) {
+      final Schema incrementingColumnSchema = schema.field(incrementingColumn).schema();
+      final Object incrementingColumnValue = record.get(incrementingColumn);
+      if (incrementingColumnValue == null) {
+        throw new ConnectException("Null value for incrementing column of type: " + incrementingColumnSchema.type());
+      } else if (isIntegralPrimitiveType(incrementingColumnValue)) {
+        extractedId = ((Number) incrementingColumnValue).longValue();
+      } else if (incrementingColumnSchema.name() != null && incrementingColumnSchema.name().equals(Decimal.LOGICAL_NAME)) {
+        final BigDecimal decimal = ((BigDecimal) incrementingColumnValue);
+        if (decimal.compareTo(LONG_MAX_VALUE_AS_BIGDEC) > 0) {
+          throw new ConnectException("Decimal value for incrementing column exceeded Long.MAX_VALUE");
+        }
+        if (decimal.scale() != 0) {
```

Comment:
An alternative to these restrictions would be to just make `TimestampIncrementingOffset` use a `BigDecimal` and convert the other types (or even make it a composite where you can choose the `long` or the `BigDecimal`. Any reason not to be a bit more liberal by doing that? Performance is the only reason I can think of.